arviz/plots/parallelplot.py

```diff
@@ -29,23 +29,32 @@ def parallelplot(trace, varnames=None, figsize=None, textsize=None, legend=True,
     shadend : float
         Alpha blending value for non-divergent points, between 0 (invisible) and 1 (opaque).
         Defaults to .025
-    skip_first : int, optional
-        Number of first samples not shown in plots (burn-in).
     ax : axes
         Matplotlib axes.
 
     Returns
     -------
     ax : matplotlib axes
     """
-    divergent = get_stats(trace[skip_first:], 'diverging')
-    trace = trace_to_dataframe(trace[skip_first:])
-    varnames = get_varnames(trace, varnames)
 
-    if len(varnames) < 2:
-        raise ValueError('This plot needs at least two variables')
+    # Get posterior samples from InferenceData
+    posterior_data = convert_to_dataset(data, group='posterior')
+    posterior, _var_names,  = [], []
+
+    # Merge chains and variables
+    for var_name, selection, _posterior in xarray_var_iter(posterior_data, var_names=var_names,
+                                                           combined=True):
+        posterior.append(_posterior.flatten())
+        _var_names.append(make_label(var_name, selection))
+    posterior = np.array(posterior)
 
-    trace = trace[varnames]
+    # Merge chains and to find divergent draws
+    divergent_data = convert_to_dataset(data, group='sample_stats')
+    divergent = xarray_var_iter(divergent_data, var_names=('diverging',), combined=True)
+    diverging_mask = next(divergent)[2].flatten()
```

Comment:
` next(divergent)[2].flatten()` looks super dangerous, like it might throw either a `StopIterationError` or an `IndexError`.

Reply:
oh, i see now - `divergent` should have length 1 as an iterator, since the chains are combined, and the `[2]` element will be the actual array.  Maybe a comment about that, then?